service/service.go

```diff
@@ -64,6 +69,11 @@ func newService(set *settings) (*service, error) {
 		return nil, fmt.Errorf("failed to get logger: %w", err)
 	}
 
+	if err = srv.colTelemetry.init(set.BuildInfo, srv.telemetry.Logger, set.Config.Service.Telemetry, set.AsyncErrorChannel); err != nil {
+		return nil, fmt.Errorf("failed to initialize telemetry: %w", err)
+	}
+	srv.telemetry.MeterProvider = srv.colTelemetry.mp
```

Comment:
Can we return this from `colTelemetry.init()`? Generally, I think we should avoid working with the private fields of `colTelemetry` here. Perhaps to enforce it move `colTelemetry` to a separate package.

Reply:
Will do that, when moving to `service/telemetry` package, in a soon followup PR.